packages/commonwealth/client/scripts/views/components/edit_profile.tsx

```diff
@@ -0,0 +1,414 @@
+      }
+    }
+    setLoading(false);
+  };
+
+  const updateProfile = async (profileUpdate: any) => {
+    try {
+      const { result } = await $.post(`${app.serverUrl()}/updateProfile/v2`, {
+        profileId: profile.id,
+        ...profileUpdate,
+        jwt: app.user.jwt,
+      });
+
+      if (result?.status === 'Success') {
+        setTimeout(() => {
+          // refresh profiles in store
+          addresses.forEach((a) => {
+            app.newProfiles.updateProfileForAccount(a.address, profileUpdate);
+          });
+          setLoading(false);
+          navigate(`/profile/id/${profile.id}`);
+        }, 1500);
+      }
+    } catch (err) {
+      setTimeout(() => {
+        setLoading(false);
+        notifyError(err.responseJSON?.error || 'Something went wrong.');
+      }, 1500);
+    }
+  };
+
+  const checkForUpdates = () => {
+    const profileUpdate: any = {};
+
+    if (!_.isEqual(name, profile?.name) && name !== '')
+      profileUpdate.name = name;
+
+    if (!_.isEqual(email, profile?.email) && email !== '')
+      profileUpdate.email = email;
+
+    // if (!_.isEqual(this.bio.textContentsAsString, this.profile?.bio))
+    //   this.profileUpdate.bio = this.bio.textContentsAsString;
+
+    if (!_.isEqual(avatarUrl, profile?.avatarUrl))
+      profileUpdate.avatarUrl = avatarUrl;
+
+    if (!_.isEqual(socials, profile?.socials))
+      profileUpdate.socials = JSON.stringify(socials);
+
+    if (!_.isEqual(backgroundImageRef.current, profile?.backgroundImage))
+      profileUpdate.backgroundImage = JSON.stringify(
+        backgroundImageRef.current
+      );
+
+    if (Object.keys(profileUpdate)?.length > 0) {
+      updateProfile(profileUpdate);
+    } else {
+      setTimeout(() => {
+        setLoading(false);
+        navigate(`/profile/id/${profile.id}`);
+      }, 1500);
+    }
+  };
+
+  const handleSaveProfile = () => {
+    setLoading(true);
+    if (!name) {
+      setDisplayNameValid(false);
+      notifyError('Please fill all required fields.');
+      setLoading(false);
+      return;
+    }
+
+    checkForUpdates();
+  };
+
+  React.useEffect(() => {
+    if (!app.isLoggedIn()) {
+      navigate(`/profile/id/${props.profileId}`);
+    }
+
+    if (props.profileId) {
+      getProfile(props.profileId);
+    }
+  }, []);
+
+  if (loading) {
+    return (
+      <div className="EditProfile full-height">
+        <div className="loading-spinner">
+          <CWSpinner />
+        </div>
+      </div>
+    );
+  }
+
+  if (error === EditProfileError.NoProfileFound) {
+    return <PageNotFound message="We cannot find profile." />;
+  }
+
+  if (error === EditProfileError.None) {
+    if (!isOwner) {
+      navigate(`/profile/id/${props.profileId}`);
+    }
+
+    // need to create an account to pass to AvatarUpload to see last upload
+    // not the best solution because address is not always available
+    // should refactor AvatarUpload to make it work with new profiles
+    let account: Account | null;
```

Comment:
you create the account on every rerender, I don't think this is intended behaviour. This kind of stuff should happen in useEffect